federationclient.go

```diff
@@ -147,3 +147,14 @@ func (ac *FederationClient) LookupStateIDs(s ServerName, roomID, eventID string)
 	err = ac.doRequest(req, &res)
 	return
 }
+
+// LookupRoomAlias looks up a room alias hosted on the remote server.
+// The domain part of the roomAlias must match the name of the server it is
+// being looked up on.
```

Comment:
If the room alias does not exist on this server, what happens?
 - An error is returned? What does that look like?
 - A  `RespDirectory` is returned which has `len(servers) == 0`?